src/benchmarks/micro/libraries/System.Numerics.Tensors/Perf_SignedIntegerTensorPrimitives.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+using System.Linq;
+
+namespace System.Numerics.Tensors.Tests
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.SIMD, Categories.JIT)]
+    [GenericTypeArguments(typeof(sbyte))]
+    [GenericTypeArguments(typeof(short))]
+    [GenericTypeArguments(typeof(int))]
+    [GenericTypeArguments(typeof(long))]
+    public class Perf_SignedIntegerTensorPrimitives<T>
+        where T : unmanaged, IBinaryInteger<T>, IMinMaxValue<T>
+    {
+        [Params(3, 64 + 7, 128 + 7, 256 + 7, 512 + 7, 3 * 512, 6 * 512)]
+        public int BufferLength;
+
+        private const int ShiftAmount = 3;
+
+        private T[] _source1;
+        private T[] _source2;
+        private T _scalar1;
+        private T[] _ones;
+        private T[] _destination;
+
+        [GlobalSetup]
+        public void Init()
+        {
+            _source1 = ValuesGenerator.Array<T>(BufferLength, seed: 42);
+            _source2 = ValuesGenerator.Array<T>(BufferLength, seed: 43);
+            _scalar1 = ValuesGenerator.Value<T>(seed: 44);
+            _ones = Enumerable.Repeat(T.One, BufferLength).ToArray();
+            _destination = new T[BufferLength];
+        }
+
+        #region Unary Operations
```

Comment:
It'd be good to go through the set of APIs and have one for each of the core work horses we have.  I think that would mean around 10 different APIs or so.  We should also ensure we have one or two of the trig methods (which could be part of the former set).